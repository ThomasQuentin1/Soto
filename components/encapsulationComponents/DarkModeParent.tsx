import React from "react";
// import NoSsr from "@material-ui/core/NoSsr";
import {
  ThemeProvider as ThemeProviderMui,
  Theme,
} from "@material-ui/core/styles";
import { ThemeProvider as ThemeProviderMainStyle } from "styled-components";
import { useDarkMode } from "../settings/useDarkMode";
import muiDarkTheme from "../../themes/mui/mainDarkTheme";
import muiLightTheme from "../../themes/mui/mainLightTheme";
import GlobalStyles from "../../styles/globalStyle";
import lightTheme from "../../themes/site/lightTheme";
import darkTheme from "../../themes/site/darkTheme";

interface DarkModeParentInterface {
  theme?: string;
  children?: any;

  isInSetting?: boolean;
}

export const getColorMode = () => {
  const [theme] = useDarkMode();

  if (theme === "dark") {
    return muiDarkTheme;
  } else {
    return muiLightTheme;
  }
};

const DarkModeParent = (props: DarkModeParentInterface) => {
  const [theme] = useDarkMode();
  let realTheme = props.theme ? props.theme : theme;
  let muiThemeMode: any = null;
  let siteThemeMode: any = null;

  muiThemeMode = realTheme === "light" ? muiLightTheme : muiDarkTheme;
  siteThemeMode = realTheme === "light" ? lightTheme : darkTheme;

  return (
    // <NoSsr>
      <ThemeProviderMainStyle theme={siteThemeMode}>
        <GlobalStyles />
        <ThemeProviderMui<Theme> theme={muiThemeMode}>
          {props.children}
        </ThemeProviderMui>
      </ThemeProviderMainStyle>
    // </NoSsr>
  );
};

export default DarkModeParent;
