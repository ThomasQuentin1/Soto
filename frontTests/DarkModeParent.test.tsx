import React from 'react';
import renderer from 'react-test-renderer';
import DarkModeParent from '../components/encapsulationComponents/DarkModeParent';

test('DarkModeParent is a container that apply the theming and style to all childs, it has to be used in pages', () => {
    const component = renderer.create(
        <DarkModeParent><p>testing DarkModeParent</p></DarkModeParent>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});