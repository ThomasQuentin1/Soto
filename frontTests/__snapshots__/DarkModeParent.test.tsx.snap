// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkModeParent is a container that apply the theming and style to all childs, it has to be used in pages 1`] = `null`;
